### PR TITLE
Refactor some common logic in the cluster validation suite

### PR DESCRIPTION
**Pull request by lack**
The mechanisms for checking MC and MCP was duplicated for both the sctp
and xt_u32 tests, and will be reused for other MC-related deployments in
the future.  This abstraction makes it easier to manage and reuse these
checks.

### CODE REVIEW

File: cnf-tests/testsuites/validationsuite/cluster/validation.go [hunk 1]
```diff
@@ -40,6 +40,77 @@ func init() {
 	machineConfigPoolNodeSelector = fmt.Sprintf("node-role.kubernetes.io/%s", roleWorkerCNF)
 }
 
+type MCMatcher func(*igntypes.Config, *clientmachineconfigv1.MachineConfig) bool
```
Comment (fedepaol) [inline cnf-tests/testsuites/validationsuite/cluster/validation.go:43]: Can you put these utilities functions at the bottom of the file?

Comment (fedepaol) [inline cnf-tests/testsuites/validationsuite/cluster/validation.go:43]: Saw both the matcher rely only on ignitionConfig, I'd remove the second parameter as nobody uses it

**Review by fedepaol — COMMENTED** [review]
Thanks for doing this!

File: cnf-tests/testsuites/validationsuite/cluster/validation.go (hunk 1 above)
Comment (lack) [reply cnf-tests/testsuites/validationsuite/cluster/validation.go:43]: That's a future-facing hint to the matcher I'm about to add for my other PR, but it's true I could remove this parameter for now and re-add it in that PR.

Comment (lack) [reply cnf-tests/testsuites/validationsuite/cluster/validation.go]: Done.

**Review by SchSeba — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**